x/trigger/client/cli/tx.go

```diff
@@ -0,0 +1,240 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+// NewTxCmd is the top-level command for trigger CLI transactions.
+func NewTxCmd() *cobra.Command {
+	txCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Aliases:                    []string{"t"},
+		Short:                      "Transaction commands for the trigger module",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	txCmd.AddCommand(
+		GetCmdAddTransactionTrigger(),
+		GetCmdAddBlockHeightTrigger(),
+		GetCmdAddBlockTimeTrigger(),
+		GetCmdDestroyTrigger(),
+	)
+
+	return txCmd
+}
+
+// GetCmdAddTransactionTrigger is a command to add a trigger for a transaction event.
+func GetCmdAddTransactionTrigger() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "create-tx-trigger <event.json> <msg.json>",
+		Args:    cobra.ExactArgs(2),
+		Aliases: []string{"tx"},
+		Short:   "Creates a new trigger that fires when a tx event is detected.",
+		Long:    strings.TrimSpace(`Creates a new trigger.  This will delay the execution of the provided message until the tx event has occurred`),
+		Example: fmt.Sprintf(`$ %[1]s tx trigger create-tx-trigger event.json message.json`, version.AppName),
```

Comment:
It would be nice if you added a few more lines here with example json for the event and msg json files.  This will be very useful for anyone who is trying to get a feel for what the arguments for the command will look like.